core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -72,12 +77,27 @@ public UnsafeSorterSpillReader(
       bufferSizeBytes = DEFAULT_BUFFER_SIZE_BYTES;
     }
 
+    try (InputStream bs = new NioBufferedFileInputStream(file, (int) bufferSizeBytes);
```

Comment:
Please add a comment here to say we don't need to hold the file open until we actually want to load the records.